### PR TITLE
fix(ios) add to validTextTracks only if we successfully insertTimeRage

### PR DESCRIPTION
**Pull request by gkueny**
insertTimeRage can fail & if we add failed textTrack to our validTextTracks array, video can crash later on selectTextTrack
we also add en empty textTrack only we we have validTextTrack

related to https://github.com/react-native-video/react-native-video/issues/3480

## Summary

Error https://github.com/react-native-video/react-native-video/issues/3480 with sideLoaded textTrack happen because 

```swift
textCompTrack.insertTimeRange
```

fail, but we already have added textTrack to our `validTextTracks` array.
Later, when we `setSelectedTextTrack` this cause video to crash.

### Motivation

This change do not fix the side loaded textTrack issue, but prevent video crash because of them

### Changes

- append `textTrack` to  `validTextTracks`  only if `textCompTrack.insertTimeRange` run successfully.
- avoid add empty `textTrack` if we have no `validTextTracks` handled (act the same if use give no `textTracks`) 
- print error to give some hint to developer

## Test plan

- Use https://github.com/gkueny/VideoErrorThrownOnSelectedTrackReproduction
- Install react-native-video with patch `yarn add react-native-video@https://github.com/gkueny/react-native-video.git#f0e32d45846003f14680a5292e42a263df2f52b3`
- Run project
- Assert that video run correctly, without (bad?) subtitle...

### CODE REVIEW

File: ios/Video/Features/RCTVideoUtils.swift [hunk 1]
```diff
@@ -299,23 +299,30 @@ enum RCTVideoUtils {
...
+                            do {
+                                try textCompTrack.insertTimeRange(
+                                    CMTimeRangeMake(start: .zero, duration: videoAsset.timeRange.duration),
+                                    of: track,
+                                    at: .zero
+                                )
+                                validTextTracks.append(textTracks[i])
+                            } catch {
+                                continue
+                            }
```
Comment (gkueny) [inline ios/Video/Features/RCTVideoUtils.swift]: one downside here, is we fail silently here. What do you recommend ?

Comment (KrzysztofMoch) [reply ios/Video/Features/RCTVideoUtils.swift]: I think we can leave TODO and later when we will "upgrade errors" we will throw here 
Eventually we cloud warn user that something is failing

**Comment by KrzysztofMoch** [comment]
You cooked 🔥 let me test it 👀

**Review by KrzysztofMoch — APPROVED** [review]
Tested and It's working 🚀
Thank You!

**Comment by gkueny** [comment]
> Tested and It's working 🚀 Thank You!

Just to be sure that there's no misunderstanding, my PR do not really fix https://github.com/react-native-video/react-native-video/issues/3480#issuecomment-1976149056 because sideloaded subtitle will not be shown to user

I just prevented the crash 


I still haven't found out why `textCompTrack.insertTimeRange` throw an error 😞

**Pull request merged**